### DISCUSSION TITLE
Nuxt hook not fired

### DISCUSSION THREAD

**Discussion by learntheropes** [nuxt/nuxt#23094] [Questions]
I have a plugin with:
```js
export default defineNuxtPlugin(async nuxtApp => {

  let commitBlockHeight;
  nuxtApp.hook('ready', async () => {
    console.log('Fetch block height')
    const [{ height }] = await $fetch('https://mempool.space/api/v1/blocks');
    commitBlockHeight = height;
  });

  return {
    provide: {
      commitBlockHeight
    }
  }
});
```

And a page with:
```vue
<script setup>
  const { $commitBlockHeight } = useNuxtApp();
</script>

<template>
  <div>{{ $commitBlockHeight }}</div>
</template>
```

However, the variable is undefined and also the console log isn't logged.
Simply, the hook is not fired.
What could be the issue?

I have a reproduction [here](https://stackblitz.com/edit/nuxt-starter-sglj8y?file=app.vue,plugins%2Ftest.js), but here obviously doesn't work since I can't run nuxt build and nuxt start

**Comment by mathix420** [comment]
I believe all [build time Nuxt Hooks](https://nuxt.com/docs/api/advanced/hooks#nuxt-hooks-build-time) are only available for modules to listen to.
